src/containers/Profile/components/EditForm.js

```diff
@@ -83,642 +86,673 @@ const starSigns = [
-        <FormField fullWidth>
-          <Label
-            htmlFor={input.name}
-            text="Email"
-            error={meta.touched && get(meta, 'error.length', 0) > 0}
+                ))
+              }
+            </FieldArray>
+          </Box>
+          <FormSpy subscription={{ values: true }}>
+            {({ values: { interests } }) => {
+              return (
+                <Collapsible
+                  trigger={({ isOpen, toggle }) => {
+                    if (isOpen) return false;
+
+                    return <EmptyStateButton onClick={toggle} text="Add interests" />;
+                  }}
+                >
+                  {({ isOpen, toggle }) => {
+                    if (!isOpen) return false;
+
+                    return (
+                      <Field
+                        name={`interests[${interests.length + 1}]`}
+                        render={({ input, meta }) => {
+                          const { onChange, ...rest } = input;
+                          const customOnChange = event => {
+                            toggle();
+                            onChange(event);
+                          };
+                          const inputProps = { onChange: customOnChange, ...rest };
+
+                          return (
+                            <FormField fullWidth>
+                              <Select
+                                loadOptions={onSearchInterests}
+                                maxLength={30}
+                                options={data.interests}
+                                placeholder="Add Interests"
+                                {...inputProps}
+                                canCreateOptions
+                              />
+                            </FormField>
+                          );
+                        }}
+                      />
+                    );
+                  }}
+                </Collapsible>
+              );
+            }}
+          </FormSpy>
+        </Section>
+
+        <Field
+          name="neighborhood"
+          render={({ input, meta }) => (
+            <FormField fullWidth>
+              <Label
+                htmlFor={input.name}
+                text="Neighborhood"
+                error={meta.touched && get(meta, 'error.length', 0) > 0}
+              />
+              <Select
+                id={input.name}
+                loadOptions={onSearchNeighborhoods}
+                options={data.neighborhoods}
+                placeholder="Neighborhood"
+                {...input}
+                canCreateOptions
+              />
+            </FormField>
+          )}
+        />
+
+        <InputGroup>
+          <Field
+            name="birthday.month"
+            render={({ input, meta }) => {
+              const { onChange, ...rest } = input;
+              const customOnChange = event => {
+                this.setState({ daysPerMonth: getDaysPerMonth(event.value) });
```

Comment:
Why are you updating state here? The `daysPerMonth` [calculated field](https://codesandbox.io/s/oq52p6v96y) should be taking care of updates for you.